internal/artifactory/aql_test.go

```diff
@@ -63,12 +71,20 @@ func TestAccSearchAQL(t *testing.T) {
 	t.Parallel()
 
 	vars, _ := EnsureArtifactoryEnvVars(t)
+	var client *Client
 
-	client := NewClient(
-		WithHost("https://artifactory.hashicorp.engineering/artifactory"),
-		WithUsername(vars["username"]),
-		WithToken(vars["token"]),
-	)
+	if vars["username"] == "" {
```

Comment:
You can probably simplify this a bit by creating a slice with the shared options, e.g.
```go
opts := []Opt{
  WithHost("https://artifactory.hashicorp.engineering/artifactory"),
  WithToken(vars["token"]),
}
if vars["username"] != nil {
  opts = append(opts, WithUsername(vars["username"]))
}
client = NewClient(opts...)
```